core/build.gradle.kts

```diff
@@ -36,7 +36,13 @@ dependencies {
   implementation(libs.commons.collections4)
   implementation(libs.guava)
   implementation(libs.h2db)
+  implementation(libs.jackson.datatype.jdk8)
```

Comment:
`jackson` package is  introduced by `LineageLogSinker` to deserializate run event to string, this will add extra dependences for all catalog and may introduce some package conflict for `jackson`.   Do you think is it necessary to move `LineageLogSinker` to server package?  @jerryshao